emmett/tools/auth/ext.py

```diff
@@ -189,13 +189,19 @@ def _inject_pipe(self):
         if self.config.inject_pipe:
             self.app.pipeline.append(self.auth.pipe)
 
-    def use_database(self, db, user_model=None):
+    def _set_model_config(self, config_name, model):
+        if model:
+            if issubclass(model, AuthModel) or issubclass(model, TimestampedModel):
+                self.config.models[config_name] = model
+            else:
+                raise RuntimeError(f'{user_model.__name__} is an invalid user model')
```

Comment:
Since `auth` module expects fields to be present in membership, group and permission models (eg: https://github.com/emmett-framework/emmett/blob/master/emmett/tools/auth/apis.py#L95-L148), I would make the check more strict:

```suggestion
    def _set_model_config(self, config_name, model):
        if not model:
            return
        _model_bases = {
            'user': AuthModel,
            'group': AuthGroup,
            'membership': AuthMembership,
            'permission': AuthPermission
        }
        if not issubclass(model, _model_bases[config_name]):
            raise RuntimeError(f'{model.__name__} is an invalid {config_name} auth model')
        self.config.models[config_name] = model
```